models/reviews-model.js

```diff
@@ -1,42 +1,104 @@
 const db = require("../db/connection");
 
-exports.selectReviews = () => {
-  return db.query("SELECT * FROM reviews;").then((data) => {
-    return data.rows;
-  });
+exports.selectReviews = (sort_by = "created_at", order = "DESC", category) => {
+  const validCategories = [
+    "strategy",
+    "hidden-roles",
+    "dexterity",
+    "push-your-luck",
+    "roll-and-write",
+    "deck-building",
+    "engine-building",
+    "jigsaw",
+  ];
+  const validOrder = ["ASC", "DESC"];
+  if (category) {
+    if (!validCategories.includes(category)) {
+      return Promise.reject({
+        status: 400,
+        msg: "category does not exist",
+      });
+    }
+  }
+  if (category) {
+    return db
+      .query(
+        `
+      SELECT reviews.*, COUNT(comments.review_id)::INT AS comment_count
+      FROM reviews
+      LEFT JOIN comments ON comments.review_id = reviews.review_id
+      WHERE reviews.category = $1
+      GROUP BY reviews.review_id
+      ORDER BY ${sort_by} ${order}
+      `,
+        [category]
+      )
+      .then((result) => {
+        if (result.rows.length === 0) {
+          return Promise.reject({
+            status: 404,
+            msg: "sorry, no reviews found for this category",
+          });
+        }
+        return result.rows;
+      });
+  } else
+    return db
+      .query(
+        `
+    SELECT reviews.*, COUNT(comments.review_id)::INT AS comment_count
+    FROM reviews
+    LEFT JOIN comments ON comments.review_id = reviews.review_id
+    GROUP BY reviews.review_id
+    ORDER BY ${sort_by} ${order}
+    ;`
+      )
+      .then((result) => {
+        if (result.rows.length === 0) {
+          return Promise.reject({
+            status: 404,
+            msg: "sorry, no reviews found",
+          });
+        }
+        return result.rows;
```

Comment:
we can make this code a little more 'dry' instead of having 2 SQL queries we can just add to an existing one if necessary. 
https://notes.northcoders.com/courses/js-back-end/complex-queries
have a look at these notes and see if you could refactor into one SQL query

Reply:
Heather's comment still applies here: build up a single query string, and have a condition that either will or won't add the conditional WHERE clause in the notes Heather's kindly shared with you